includes/class-wc-stripe-payment-tokens.php

```diff
@@ -262,15 +262,19 @@ public function woocommerce_get_customer_upe_payment_tokens( $tokens, $user_id,
 		}
 
 		try {
-			$stored_tokens = [];
-
+			$stored_tokens     = [];
 			$deprecated_tokens = [];
 
 			foreach ( $tokens as $token ) {
 				if ( in_array( $token->get_gateway_id(), self::UPE_REUSABLE_GATEWAYS_BY_PAYMENT_METHOD, true ) ) {
 
-					// APM tokens from before Split PE was in place that will get removed.
-					if ( 'stripe' === $token->get_gateway_id() && 'sepa' === $token->get_type() ) {
+					// Remove the following deprecated tokens:
+					// - APM tokens from before Split PE was in place.
+					// - Tokens using the sources API. Payments using these will fail with the PaymentMethods API.
+					if (
+						( 'stripe' === $token->get_gateway_id() && 'sepa' === $token->get_type() ) ||
+						str_starts_with( $token->get_token(), 'src_' )
```

Comment:
Alternatively, I [evaluated](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3241/commits/1827a7b613dc639ccde4cd90d489ac5f6c590238) only removing the local sources tokens _when_ they were already migrated on the Stripe end.

I decided to take the current approach because:
- Processing a payment with a source using the PaymentMethods API will fail
- Assuming the merchant hasn't migrated their sources, the solution for the shopper would be to save the payment method again
- With the current approach, they won't see any of their saved payment methods. This isn't ideal, but probably a better experience than having their payment fail
- If they see their payment methods displayed, they won't likely try to save a new one

Happy to take a different approach!

Reply:
Hi @a-danae! Thanks for your work on this plugin. I just filed https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3498 because I think there might be a slight misunderstanding here. You say, "Processing a payment with a source using the PaymentMethods API will fail," but according to [Stripe's API Docs](https://docs.stripe.com/payments/payment-methods/transitioning#compatibility), they do support using credit card Sources with the new Payment Methods API.

Reply:
@giilby Danae no longer works on this project. I don't know exactly what she meant by that comment, however, IIRC the issue we were seeing is specifically related to SEPA. If you look at the issue this PR was created to address (https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3134) it specifically mentions SEPA. 

At the time `src_` tokens were synonymous with SEPA tokens and SEPA sources were/are failing when using the Payment methods API. While that Stripe doc you mentioned says that sources will continue to work with the Payment Methods API, the Sources Doc (https://docs.stripe.com/sources) says that Source objects for local payment methods (LPM), like SEPA, won't work, are deprecated and we should no longer generate Source objects for card payments either.  

So I think putting those two points together -- **source objects linked to cards should continue to work**, however **source objects for local methods won't**. 

We should update the code added in this PR to make sure card source objects can still be used, while source objects for LPMs don't. I'll make a comment on your issue to make sure that nuance is captured.